src/components/manage/Controlpanels/Aliases.jsx

```diff
@@ -61,635 +60,541 @@ const filterChoices = [
 
 const itemsPerPageChoices = [10, 25, 50, 'All'];
 
-/**
- * Aliases class.
- * @class Aliases
- * @extends Component
- */
-class Aliases extends Component {
-  /**
-   * Property types.
-   * @property {Object} propTypes Property types.
-   * @static
-   */
-  static propTypes = {
-    addAliases: PropTypes.func.isRequired,
-    getAliases: PropTypes.func.isRequired,
-    removeAliases: PropTypes.func.isRequired,
-  };
+const Aliases = (props) => {
+  const title = props;
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const { pathname } = useLocation();
+  const history = useHistory();
 
-  /**
-   * Constructor
-   * @method constructor
-   * @param {Object} props Component properties
-   * @constructs Aliases
-   */
-  constructor(props) {
-    super(props);
-    this.state = {
-      isClient: false,
-      filterType: filterChoices[0],
-      createdBefore: null,
-      altUrlPath: '',
-      isAltUrlCorrect: false,
-      targetUrlPath: '',
-      isTargetUrlCorrect: false,
-      aliasesToRemove: [],
-      errorMessageAdd: '',
-      filterQuery: '',
-      aliases: [],
-      activePage: 1,
-      pages: '',
-      itemsPerPage: 10,
-    };
-  }
+  const aliases = useSelector((state) => state.aliases);
+  const [filterType, setFilterType] = useState(filterChoices[0]);
+  const [createdBefore, setCreatedBefore] = useState(null);
+  const [altUrlPath, setAltUrlPath] = useState('');
+  const [isAltUrlCorrect, setIsAltUrlCorrect] = useState(false);
+  const [targetUrlPath, setTargetUrlPath] = useState('');
+  const [isTargetUrlCorrect, setIsTargetUrlCorrect] = useState(false);
+  const [aliasesToRemove, setAliasesToRemove] = useState([]);
+  const [errorMessageAdd, setErrorMessageAdd] = useState('');
+  const [filterQuery, setFilterQuery] = useState('');
+  const [activePage, setActivePage] = useState(1);
+  const [pages, setPages] = useState('');
+  const [itemsPerPage, setItemsPerPage] = useState(10);
+  const isClient = useClient();
 
-  /**
-   * Component did mount
-   * @method componentDidMount
-   * @returns {undefined}
-   */
-  componentDidMount() {
-    const { filterQuery, filterType, createdBefore, itemsPerPage } = this.state;
-    this.setState({ isClient: true });
-    this.props.getAliases(getBaseUrl(this.props.pathname), {
-      query: filterQuery,
-      manual: filterType.value,
-      datetime: createdBefore,
-      batchSize: itemsPerPage,
-    });
-  }
+  useEffect(() => {
+    dispatch(
```

Comment:
@Tishasoumya-02 I get duplicate calls to @aliases in /controlpanel/aliases. This is due to multiple useEffect calls. The useEffect can itself serve purpose of all lifecycle methods of class based components. In many cases, there can be one useEffect for componentDidMount and componentDidUpdate.